plutus-tx/src/PlutusTx/Builtins/HasOpaque.hs

```diff
@@ -274,7 +274,11 @@ class MkNil arep where
 instance MkNil BuiltinInteger
 instance MkNil BuiltinBool
 instance MkNil BuiltinData
-instance MkNil (BuiltinPair BuiltinData BuiltinData)
+-- TODO: the following two instances are not implemented in the plugin.
+-- They require changes to PlutusTx.Compiler.Expr.compileExpr
+-- See https://github.com/IntersectMBO/plutus-private/issues/1604
+instance (MkNil a) => MkNil (BuiltinList a)
+instance (MkNil a, MkNil b) => MkNil (BuiltinPair a b)
```

Comment:
Can we remove them if they aren't gonna work anyway? If that requires removing broken code as well, I'm OK with that. 

@zliu41 what do you think?

Reply:
Remove those functions then too, they're broken anyway. Note that if you address the other comments such as introducing `MkNil (BuiltinPair a b)` constraints, fewer of the functions will remain broken.